File: Remove_Colored_Pieces_if_Both_Neighbors_are_the_Same_Color.py
class Solution:
    def winnerOfGame(self, colors: str) -> bool:
        if len(colors) <= 2:
            return False

        alice_turn = 0
        bob_turn = 0
        
        for i in range(1, len(colors) - 1):
            if colors[i - 1] == colors[i] == colors[i + 1]:
                if colors[i] == "A":
                    alice_turn += 1
                else:
                    bob_turn += 1
        
        return alice_turn > bob_turn

File: Remove_Colored_Pieces_if_Both_Neighbors_are_the_Same_Color.java
class Solution {
    public boolean winnerOfGame(String colors) {
        if (colors.length() <= 2) {
            return false;
        }

        int aliceTurn = 0;
        int bobTurn = 0;

        for (int i = 1; i < colors.length() - 1; i++) {
            if (colors.charAt(i - 1) == colors.charAt(i) && colors.charAt(i) == colors.charAt(i + 1)) {
                if (colors.charAt(i) == 'A') {
                    aliceTurn++;
                } else {
                    bobTurn++;
                }
            }
        }

        return aliceTurn > bobTurn;        
    }
}
